x/mongo/driver/topology/fsm.go

```diff
@@ -88,23 +92,24 @@ func (f *fsm) apply(s description.Server) (description.Topology, error) {
 		}
 	}
 
+	updatedDesc := s
```

Comment:
I changed all of the `applyTo*` functions to return a `description.Server`. The only one that can potentially return an updated description is `applyToSingle` so technically all of the others could have kept their old signature, but it made sense for consistency and encapsulates that spec detail into `applyToSingle` rather than having the caller be aware of this.